salt/_pillar/metalk8s_solutions.py

```diff
@@ -29,10 +29,17 @@ def _load_solutions(bootstrap_id):
 
     errors = []
     try:
-        result['available'] = __salt__['saltutil.cmd'](
+        available_ret = __salt__['saltutil.cmd'](
             tgt=bootstrap_id,
             fun='metalk8s_solutions.list_available',
-        )[bootstrap_id]['ret']
+        )[bootstrap_id]
+        if available_ret['retcode'] != 0:
+            raise Exception('[{}] {}'.format(
+                available_ret['retcode'],
+                available_ret['ret']
+            ))
```

Comment:
I am not sure we want to raise like that ... no>>
why not something like :
```
return __utils__['pillar_utils.errors_to_dict']([
    '{}'.format(available_ret['ret'])
])
```

Reply:
If we raise that will be handle by the except block which is good IMO, we will have something like 
```
Error when listing available Solutions:[<error_code>] <error_msg>
```